src/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContext.cs

```diff
@@ -0,0 +1,317 @@
+                {
+                    Debug.WriteLineIf(s_runtimeLicenseContextSwitch.TraceVerbose, "licenseFile: " + licenseFile.ToString());
+                    Debug.WriteLineIf(s_runtimeLicenseContextSwitch.TraceVerbose, "opening licenses file over URI " + licenseFile.ToString());
+                    Stream s = OpenRead(licenseFile);
+                    if (s != null)
+                    {
+                        string[] segments = licenseFile.Segments;
+                        string licFileName = segments[segments.Length - 1];
+                        string key = licFileName.Substring(0, licFileName.LastIndexOf("."));
+                        DesigntimeLicenseContextSerializer.Deserialize(s, key.ToUpper(CultureInfo.InvariantCulture), this);
+                    }
+                }
+            }
+            Debug.WriteLineIf(s_runtimeLicenseContextSwitch.TraceVerbose, "returning : " + (string)savedLicenseKeys[type.AssemblyQualifiedName]);
+            return (string)savedLicenseKeys[type.AssemblyQualifiedName];
+        }
+
+        /**
+        * Looks up a .licenses file in the assembly manifest using 
+        * case-insensitive lookup rules.  We do this because the name
+        * we are attempting to locate could have different casing 
+        * depending on how the assembly was loaded.
+        **/
+        private Stream CaseInsensitiveManifestResourceStreamLookup(Assembly satellite, string name)
+        {
+            CompareInfo comparer = CultureInfo.InvariantCulture.CompareInfo;
+
+            //loop through the resource names in the assembly
+            // we try to handle the case where the assembly file name has been renamed
+            // by trying to guess the original file name based on the assembly name...
+            string assemblyShortName = satellite.GetName().Name;
+            foreach (string existingName in satellite.GetManifestResourceNames())
+            {
+                if (comparer.Compare(existingName, name, CompareOptions.IgnoreCase) == 0 ||
+                    comparer.Compare(existingName, assemblyShortName + ".exe.licenses") == 0 ||
+                    comparer.Compare(existingName, assemblyShortName + ".dll.licenses") == 0)
+                {
+                    name = existingName;
+                    break;
+                }
+            }
+
+            //finally, attempt to return our stream based on the 
+            //case insensitive match we found
+            //
+            return satellite.GetManifestResourceStream(name);
+        }
+
+        private static Stream OpenRead(Uri resourceUri)
+        {
+            Stream result = null;
+
+            //TODO NETSTANDARD2.0 - review this
+            //PermissionSet perms = new PermissionSet(PermissionState.Unrestricted);
+
+            //perms.Assert();
+            try
+            {
+                WebClient webClient = new WebClient();
+                webClient.Credentials = CredentialCache.DefaultCredentials;
```

Comment:
We should test this and make sure it behaves as expected (e.g. the credentials are a nop) on Unix.